### ISSUE TITLE
Inference baichuan2-7b-chat with fp8 and int5 failed

### ISSUE THREAD

**Issue by Fred-cell** [intel-analytics/BigDL#9680]
inference baichuan2-7b-chat with int5 and fp8 failed.
2023-12-13 19:17:44,210 - transformers_modules.Baichuan2-7B-Chat.modeling_baichuan - WARNING - Xformers is not installed correctly. If you want to use memory_efficient_attention to accelerate training use the following command to install Xformers
pip install xformers.
2023-12-13 19:18:00,751 - bigdl.llm.transformers.utils - INFO - Converting the current model to fp8 format......
2023-12-13 19:18:08,788 - bigdl.llm.transformers.models.baichuan2 - WARNING - Xformers is not installed correctly. If you want to use memory_efficient_attention to accelerate training use the following command to install Xformers
pip install xformers.
<class 'transformers_modules.Baichuan2-7B-Chat.modeling_baichuan.BaichuanForCausalLM'>
Traceback (most recent call last):
  File "/home/fred/LLM/text-generation/bigdl-llm/BigDL/python/llm/example/GPU/HF-Transformers-AutoModels/Model/baichuan2/generate.py", line 65, in <module>
    output = model.generate(input_ids,
  File "/root/anaconda3/envs/bigdl-llm/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/home/fred/LLM/text-generation/bigdl-llm/BigDL/python/llm/example/GPU/HF-Transformers-AutoModels/Model/baichuan2/benchmark_util.py", line 1611, in generate
    return self.sample(
  File "/home/fred/LLM/text-generation/bigdl-llm/BigDL/python/llm/example/GPU/HF-Transformers-AutoModels/Model/baichuan2/benchmark_util.py", line 2718, in sample
    next_tokens = torch.multinomial(probs, num_samples=1).squeeze(1)
RuntimeError: probability tensor contains either `inf`, `nan` or element < 0

**Comment by Ariadne330** [comment]
Hi @Fred-cell 
We have reproduced your issue and found the problem is caused by using fused mlp layers in Baichuan2. 
To address this issue, you can remove the `convert_forward` related codes in `/<your env path>/site-packages/bigdl/llm/transformers/convert.py` like this:
```python
    elif model.config.model_type == "baichuan" and model.config.vocab_size == 125696:
        # baichuan2
        if model.config.hidden_size == 4096:
            # baichuan2-7B
            modeling_module_name = model.__class__.__module__
            module = importlib.import_module(modeling_module_name)
            from bigdl.llm.transformers.models.baichuan2 import baichuan_attention_forward_7b
            # from bigdl.llm.transformers.models.baichuan2 import baichuan_mlp_forward
            convert_forward(model,
                            module.Attention,
                            baichuan_attention_forward_7b
                            )
            convert_forward(model,
                            module.RMSNorm,
                            llama_rms_norm_forward)
            # convert_forward(model,
            #                 module.MLP,
            #                 baichuan_mlp_forward)
        elif model.config.hidden_size == 5120:
            # baichuan2-13B
            modeling_module_name = model.__class__.__module__
            module = importlib.import_module(modeling_module_name)
            from bigdl.llm.transformers.models.baichuan2 import baichuan_attention_forward_13b
            from bigdl.llm.transformers.models.baichuan2 import baichuan_13b_rms_norm_forward
            # from bigdl.llm.transformers.models.baichuan2 import baichuan_mlp_forward
            convert_forward(model,
                            module.BaichuanAttention,
                            baichuan_attention_forward_13b
                            )
            # baichuan2-13B's RMSNorm is a little different
            convert_forward(model,
                            module.RMSNorm,
                            baichuan_13b_rms_norm_forward)
            # convert_forward(model,
            #                 module.MLP,
            #                 baichuan_mlp_forward)

```


Or you can just downgrade to **bigdl-llm[xpu] 2.5.0b20231206** to produce the right result.

**Comment by hkvision** [comment]
@MeouSker77 Please check fused mlp layers in baichuan.

**Comment by MeouSker77** [comment]
fix in #9689
